src/main/java/Duke.java

```diff
@@ -1,10 +1,178 @@
+            } else {
+                String output = LINES + " " + s + "\n" + LINES;
+                System.out.println(output);
+            }
+        }
+    }
+
+    //list function
+    public static void list(Task[] tasks) {
+        System.out.print(LINES);
+        for (int i = 0; tasks[i] != null; i++) {
+            int j = i + 1;
+            System.out.println(j + "." + tasks[i]);
+        }
+        System.out.print(LINES);
+    }
+
+    //Marks a specified task input as done
+    public static void markDone(String input, Task[] tasks, int taskCount) {
+        //isolate 'x' from 'done x', where x is a number
+        int index = Integer.parseInt(input.substring(5));
+        if (index > taskCount) {
+            System.out.println(LINES + NO_TASK_MESSAGE + LINES);
+        }
+        else {
+            tasks[index - 1].markAsDone();
+            System.out.print(LINES + TASK_COMPLETE_MESSAGE + tasks[index-1] + "\n" + LINES);
+        }
+    }
+    //Adds task to tasks
+    public static int addTask(Task[] tasks, String input, int taskCount) {
+        if (input.toUpperCase().startsWith("TODO ")) {
+            tasks[taskCount] = new Todo(input.substring(5));
+        }
+        else if (input.toUpperCase().startsWith("DEADLINE ")) {
+            int slashIndex = input.indexOf("/");
+            String task = input.substring(9, (slashIndex-1));
+            String dueDate = input.substring(slashIndex+4);
+            tasks[taskCount] = new Deadline(task, dueDate);
+        }
+        else {
+            int slashIndex = input.indexOf("/");
+            String task = input.substring(6, (slashIndex-1));
+            String timeRange = input.substring(slashIndex+4);
+            tasks[taskCount] = new Event(task, timeRange);
+        }
+        System.out.println(LINES + ADD_MESSAGE + tasks[taskCount]);
+        taskCount += 1;
+        System.out.println(" Now you have " + taskCount + " tasks in the list.\n" + LINES);
+        return taskCount;
+    }
+
+    //Clears all tasks
+    public static void initJim() {
+        tasks = new Task[100];
+        taskCount = 0;
+    }
+
+    public static String getUserInput() {
+        System.out.print(" How can I help you: ");
+        return sc.nextLine();
+    }
+
+    public static void genshinHelper() {
+        int totalMats = 0;
+        System.out.print(LINES + " How many Top Tier: ");
+        String top = sc.nextLine();
+        totalMats += Integer.parseInt(top)*27;
+        System.out.print(" How many High Tier: ");
+        String high = sc.nextLine();
+        totalMats += Integer.parseInt(high)*9;
+        System.out.print(" How many Mid Tier: ");
+        String mid = sc.nextLine();
+        totalMats += Integer.parseInt(mid)*3;
+        System.out.print(" How many Low Tier: ");
+        String low = sc.nextLine();
+        totalMats += Integer.parseInt(low);
+        System.out.println(" Total low tier mats required: " + totalMats + "\n" + LINES);
+    }
```

Comment:
Could consider providing documentation for what the function does to allow for ease of understanding for future collaboration.
Could consider giving meaningful names for each of the numbers 3, 9 and 27